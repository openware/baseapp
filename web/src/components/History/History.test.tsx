import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { History, HistoryProps } from '.';
import { CellData } from '../';

const data: CellData[][] = [
    ['10:40', 'Market', 'BTC/USDT', 'Buy', '9.400,0', '0, 4005', '3.459'],
    ['10:40', 'Market', 'BTC/USDT', 'Buy', '9.400,0', '0, 4005', '3.459'],
    ['10:40', 'Market', 'BTC/USDT', 'Buy', '9.400,0', '0, 4005', '3.459'],
];

const defaultProps: HistoryProps = {
    data,
};

const setup = (props: Partial<HistoryProps> = {}) =>
    shallow(<History {...{ ...defaultProps, ...props }} />);

describe('History', () => {
    let wrapper: ShallowWrapper<History>;

    beforeEach(() => {
       wrapper = setup();
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render custom action cell', () => {
        const renderedBuyAction = (
            <span className="cr-history-action cr-history-action--buy">bid</span>
        );

        const renderedSellAction = (
            <span className="cr-history-action cr-history-action--sell">ask</span>
        );

        const instance = wrapper.instance() as History;

        expect(instance.renderAction('bid')).toEqual(renderedBuyAction);
        expect(instance.renderAction('ask')).toEqual(renderedSellAction);
    });
});
