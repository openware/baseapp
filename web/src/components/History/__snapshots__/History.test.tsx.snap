// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History should render 1`] = `
<Component
  data={
    Array [
      Array [
        "10:40",
        <span
          className="cr-history-action"
        >
          market
        </span>,
        "BTC/USDT",
        "Buy",
        "9.400,0",
        "0, 4005",
        "3.459",
      ],
      Array [
        "10:40",
        <span
          className="cr-history-action"
        >
          market
        </span>,
        "BTC/USDT",
        "Buy",
        "9.400,0",
        "0, 4005",
        "3.459",
      ],
      Array [
        "10:40",
        <span
          className="cr-history-action"
        >
          market
        </span>,
        "BTC/USDT",
        "Buy",
        "9.400,0",
        "0, 4005",
        "3.459",
      ],
    ]
  }
  header={
    Array [
      "Time",
      "Action",
      "Price",
      "Amount",
      "Total",
    ]
  }
  titleComponent="Trades History"
/>
`;
